api/repositories/org_repository.go

```diff
@@ -17,14 +19,16 @@ import (
 	"sigs.k8s.io/hierarchical-namespaces/api/v1alpha2"
 )
 
-//+kubebuilder:rbac:groups=hnc.x-k8s.io,resources=subnamespaceanchors,verbs=list;create;watch
+//+kubebuilder:rbac:groups=hnc.x-k8s.io,resources=subnamespaceanchors,verbs=list;create;delete;watch
+//+kubebuilder:rbac:groups=hnc.x-k8s.io,resources=hierarchyconfigurations,verbs=get;list;watch;update
 //+kubebuilder:rbac:groups="",resources=serviceaccounts;secrets,verbs=get;list;create;delete;watch
 
 //counterfeiter:generate -o fake -fake-name CFOrgRepository . CFOrgRepository
 
 type CFOrgRepository interface {
 	CreateOrg(ctx context.Context, info authorization.Info, org CreateOrgMessage) (OrgRecord, error)
 	ListOrgs(ctx context.Context, info authorization.Info, orgNames []string) ([]OrgRecord, error)
+	DeleteOrg(context.Context, authorization.Info, DeleteOrgMessage) error
```

Comment:
As a general rule in Go, you want to define your interfaces where they're used and not where the implementation is defined. This encourages you to keep the interface smaller, since you only add methods you use, and also makes it simpler to swap in a different implementation later.

I missed that we'd started to move in that direction here. I know you didn't introduce the pattern in this PR, but this would be a good time to move the interface or at least not consolidate on the interface in the "wrong" place.

Reply:
This is great feedback! @matt-royal 

@julian-hj and I saw that there were 2 interfaces (one is the repository and one in the handler) for the same org-repo implementation and we ended up deleting one in the handler. Looking back we should have probably looked at other repos to make a decision on which on to delete 👍🏽

Reply:
Addressed in commit - https://github.com/cloudfoundry/cf-k8s-controllers/pull/464/commits/a99f88f188236f227f4a094857f8ada66c9204e7

Reply:
@matt-royal @akrishna90 I suspect that if you just move the interface into the "api" package and use it from there, you will get a cyclic dependency error, since the "api" package already imports the "repositories". You might need to move interfaces into a separate package if we don't want them in "repositories".

Reply:
...or apparently not, since it doesn't seem as though the repositories repo was actually using that interface. So, never mind...